inst/csv/replacementPatterns.csv

```diff
@@ -18,20 +18,18 @@ oracle,"DATEADD(m,@months,@date)","ADD_MONTHS(@date, @months)"
 oracle,"DATEADD(year,@years,@date)","ADD_MONTHS(@date, 12 * @years)"
 oracle,"DATEADD(yyyy,@years,@date)","ADD_MONTHS(@date, 12 * @years)"
 oracle,"DATEADD(yy,@years,@date)","ADD_MONTHS(@date, 12 * @years)"
-oracle,"DATEDIFF(day,@start, @end)",(CAST(@end AS DATE) - CAST(@start AS DATE))
-oracle,"DATEDIFF(dd,@start, @end)",(CAST(@end AS DATE) - CAST(@start AS DATE))
-oracle,"DATEDIFF(d,@start, @end)",(CAST(@end AS DATE) - CAST(@start AS DATE))
+oracle,"DATEDIFF(day,@start, @end)",(@end - @start)

```

Comment:
Could you help me understand how this relates to the issue? The rules you changed did not use TO_CHAR. I'm pretty sure I introduced the casts to date for a valid reason (ie.  I think SQL Server auto-casts to DATE, but Oracle doesn't).

Reply:
The rules were chaining together, as I was changing how CAST * as DATE) in impacted this rule.  It was at this point that I  realized that if you are doing a DATEDIFF(), whatever @start and @end resolves to should be a DATE type.  Any SqlRender rules will apply to whatever is inside @start and @end. So there is no CAST required on Oracle (we should assume the value in @start has already been properly resolved to a date), and all you do in oracle is subtract one from the other.